src/NativePresets/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+if(NOT ENABLE_PRESETS OR NOT ENABLE_NATIVE_PRESETS)
+    return()
+endif()
+
+# Using a macro to reduce repetition.
+macro(add_native_preset name)
+    if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        add_library(${name}_static STATIC ${name}.cpp)
```

Comment:
Why are you forcing static linking instead of letting the user set [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html)?

Reply:
As stated in my other comment, the code itself is currently not ready to support proper DLL building on Windows as it's missing dllexport attributes and proper interface classes.

If you have interest in building a proper external API that hides implementation details but exports all required interfaces in a well-declared way, we can do this together with Mischa on Discord. Every input on that counts.